### ISSUE TITLE
Traversing Lst<TryAsync<T>> returns empty

### ISSUE THREAD

**Issue by NickDarvey** [louthy/language-ext#279]
**Repro**
```
List(
    TryAsync(() => 1),
    TryAsync(() => 2),
    TryAsync(() => 3),
    TryAsync(() => 4))
.Traverse(n => n)
.Match(Succ: res => Debug.Assert(res == List(1,2,3,4)),
       Fail: ex => Debug.Fail(ex.ToString()))
.GetAwaiter().GetResult();
```

**Expected**
Debug assertion passes

**Actual**
Debug assertion fails.
It matches on `Succ`, however the `Lst<int>` is empty. Its non-public members `Root` and `Value` are also empty.

**Comment by louthy** [comment]
@NickDarvey I'll take a look.  btw, you can just use `Sequence` instead of `Traverse(n => n)`.  `Sequence` is `Traverse` with identity.

**Comment by louthy** [comment]
Fixed in `v2.1.25` on nu-get now.
